packages/sdk/src/interfaces/text-to-image.ts

```diff
@@ -0,0 +1,80 @@
+import { InterfaceDeclaration } from '../integration/definition'
+import z from '../zui'
+
+const ImageModelRefSchema = z.object({
+  id: z.string(),
+})
+
+const ImageModelSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  costPerImage: z.number().describe('Cost per image generation, in U.S. dollars'),
+  sizes: z.array(z.string()).describe('Available image sizes'),
+  defaultSize: z.string().describe('Default image size generated by model'),
+})
+
+const ImageGenerationParamsSchema = z
+  .object({
+    id: z.string(), // TODO: remove
+  })
+  .describe('Model-specific parameters for image generation')
+
+const GenerateImageInputSchema = <TModelRef extends z.ZodSchema, TParams extends z.ZodSchema>(
+  imageModelRefSchema: TModelRef,
+  paramsSchema: TParams
+) =>
+  z.object({
+    model: imageModelRefSchema.optional().describe('Model to use for image generation'),
+    prompt: z.string(),
+    size: z.string().optional(),
+    params: paramsSchema.optional(),
+  })
+
+const GenerateContentInputBaseSchema = GenerateImageInputSchema(ImageModelRefSchema, ImageGenerationParamsSchema)
+
+const GenerateImageOutputSchema = z.object({
+  model: z.string().describe('Model name used'),
+  imageUrl: z.string().describe('Temporary URL of generated image'),
+})
+
+export const textToImage = new InterfaceDeclaration({
+  name: 'textToImage',
+  version: '1.0.0',
```

Comment:
**nitpick:** no strong opinion, but I've never released a full major in my life XD. I usually start with `0.0.1` as an excuse to allow me to break in the future XD... You decide

Reply:
I already deployed it to staging for testing 😅 I usually follow semver and start on 1.0.0 and go from there (major bump for breaking change, minor bump for new feature, patch bump for a fix)